tests/ui/eta.rs

```diff
@@ -220,3 +220,18 @@ impl std::ops::Deref for Bar {
 fn test_deref_with_trait_method() {
     let _ = [Bar].iter().map(|s| s.to_string()).collect::<Vec<_>>();
 }
+
+fn mutable_closure_used_twice(x: Vec<i32>, y: Vec<i32>) {
+    let mut res = Vec::new();
+    let mut add_to_res = |n| res.push(n);
+    x.into_iter().for_each(|x| add_to_res(x));
+    y.into_iter().for_each(|x| add_to_res(x));
```

Comment:
Can you add this line a third time? I expect that your changes will just work here and suggest to use `&mut` for the first 2 lines, but having this as a regression test will help in the future.